src/app/clusters/level-control/level-control.cpp

```diff
@@ -246,24 +246,27 @@ void emberAfLevelControlClusterServerTickCallback(EndpointId endpoint)
         if (state->commandId == Commands::MoveToLevelWithOnOff::Id || state->commandId == Commands::MoveWithOnOff::Id ||
             state->commandId == Commands::StepWithOnOff::Id)
         {
-            setOnOffValue(endpoint, (currentLevel != state->minLevel));
+            status = OnOff::Attributes::OnOff::Set(endpoint, currentLevel != state->minLevel);
```

Comment:
> Testing OnOffTransitionTime feature, when receiving off command, the previous order: onoff attribute change to off, then execute move-to-level,

@jmartinez-silabs @Rob-Houtepen-Signify could you please check this behavior (and this PR, which is aiming to change it) against the spec?

Reply:
You are breaking functionalities and also removing a lot of checks and validation by replacing this function call and forcing the onoff attribute to false. This is not the right way to go.

`setOnOffValue` checks if the on-off cluster is on the same endpoint if so it will call `EmberAfStatus OnOffServer::setOnOffValue` which will validate values and the feature map. it then does the behaviour for the OnTime attribute, the level transitions, links with mode Select and scenes.

Reply:
I believe your premise is wrong for the whole logic that you are trying to do. The OnOff attribute state should be turned off when receiving the off command and the level transition started. It is up to the application in my opinion to not physically turn off the light until the desired transition time is done.

I cannot find explicitly in the spec that for OnOffTransition, or Off transition that the OnOff State must be changed after the transition time.

In the spec, the on-off command just specifies that On Off state should be changed directly to false.

The OffWithEffect Command is clearer on that topic as it says 
```
7.4.3. Effect on Receipt
On receipt of the OffWithEffect command, the server SHALL check the value of the GlobalSceneControl attribute.

If the GlobalSceneControl attribute is equal to TRUE, the server SHALL store its settings in its global scene then set the GlobalSceneControl attribute to FALSE, then set the OnOff attribute to FALSE and if the OnTime attribute is supported set the OnTime attribute to 0.

If the GlobalSceneControl attribute is equal to FALSE, the server SHALL only set the OnOff attribute to FALSE.
```
This is a good example of how the effect is dealt with in the app code. The OnOff attribute is changed directly but the app needs to trigger the effect before physically closing the Light.

I'll let Rob chime about the spec.

Reply:
These attributes are a nightmare! Including the OnLevel attribute. I have no practical experience with them as they are not mandatory for lights. Which device type uses these? Can we not remove these attributes from the SDK?

Reply:
@jmartinez-silabs @Rob-Houtepen-Signify  What's your suggestion on the next step?

OnOffTransitionTime is an optional attribute defined in Level Control cluster, and might be enabled by some lighting customers, some description from Clusters Spec:
```
1.6.5.9. OnOffTransitionTime Attribute

The OnOffTransitionTime attribute represents the time taken to move to or from the target level when On or Off commands are received by an On/Off cluster on the same endpoint.
```
The behavior is clear, and the current SDK already implemented most of the OnOffTransitionTime feature, including the fade in/out logic, this PR aims to trigger the Off state change after the fade out is done, then the whole feature works as expected.

Reply:
Again, the section of the spec you show is clear on the transition of the level yes. But it doesn't say anything about when the on off attributes changes.

The command received is off. In the spec, the command OnOff says to set the attribute state to off on the command reception. Nothing is said about changing the attribute state after transition time is complete.

Did you test other use cases of the on off and level cluster with your changes?

Is the onLevel (when null or non-null) behaviour still functional for on and off?

Reply:
Yes, I had tested most of the on/off and levelcontrol cluster commands, my changes has no extra impact. And the onLevel behaviour still functional for on and off.